test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -939,16 +948,27 @@
                       (mt/formatted-rows [int int int int])
                       first))))))))
 
+(mt/defdataset diff-type-test-cases
+  [["times"
+    [{:field-name "a_text",      :base-type :type/Text}
+     {:field-name "a_datetime",  :base-type :type/DateTime}
+     {:field-name "b_text",      :base-type :type/Text}
+     {:field-name "b_time",      :base-type :type/Time}]
+    [["2022-10-02T01:00:00"    ; a_text
+      #t "2022-10-02T01:00:00" ; a_datetime
+      "05:00:00"               ; b_text
+      #t "05:00:00"]]]])       ; b_time
```

Comment:
The same suggestion I made in 

https://github.com/metabase/metabase/pull/27068#pullrequestreview-1212442994

last week applies here too... we already have `test-data-with-time` which should work here so let's avoid creating another dataset at runtime if we don't need to

Reply:
Cherry picked the commit [0152feb](https://github.com/metabase/metabase/pull/27076/commits/0152feb3b22e949c7f421965d58c9c7bf300d75c)